### DISCUSSION TITLE
Turning theme into regular values

### DISCUSSION THREAD

**Discussion by selbekk** [Shopify/restyle#202] [Ideas]
Can we implement a function that takes a set of Restyle styles (with design tokens), that turns the styles into "regular" styling values?

I.e. something like this:

```tsx
const customStyles = {
  backgroundColor: "cardPrimary",
  borderRadius: "md",
};
const regularStyles = useStyles(customStyles)
// === { backgroundColor: "#abcabc", borderRadius: 12 }
```

How could I build a hook like this in userland?

**Comment by fortmarek** [comment]
Shouldn't you be able to use the `useRestyle` hook as described [here](https://github.com/Shopify/restyle#custom-components)?

**Reply by selbekk** [reply]
Well, in that case I would need to specify what restyleFunctions I want applied, and add a lot of complexity. But perhaps that's something I could abstract away in a custom hook somehow.
